applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -110,26 +119,50 @@ static int stream_index_get(struct bt_audio_stream *stream, uint8_t *index)
 	return -EINVAL;
 }
 
-static int stream_index_from_conn_get(struct bt_conn *conn, uint8_t *index)
+static bool all_active_headsets_started(void)
 {
-	for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
-		if (audio_streams[i].conn == conn) {
-			*index = i;
-			return 0;
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (headsets[i].sink_stream != NULL &&
+		    headsets[i].sink_stream->ep->status.state != BT_AUDIO_EP_STATE_STREAMING) {
+			return false;
 		}
 	}
-
-	LOG_DBG("Stream may not have started yet");
-
-	return -EINVAL;
+	return true;
 }
 
-static int headset_conn_index_get(struct bt_conn *conn, uint8_t *index)
+/**
+ * @brief  Get channel index based on either conn or stream
+ *
+ * @param[in]  conn   The connection to search for, can be NULL
+ * @param[in]  stream The stream to search for, can be NULL
+ * @param[out] index  The channel index
+ *
+ * @return 0 if succesfull, -EINVAL if there is no match
+ */
+static int channel_index_get(struct bt_conn *conn, struct bt_audio_stream *stream, uint8_t *index)
 {
-	for (int i = 0; i < ARRAY_SIZE(headset_conn); i++) {
-		if (headset_conn[i] == conn) {
-			*index = i;
-			return 0;
+	if (conn == NULL && stream == NULL) {
+		LOG_ERR("No conn or stream provided");
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (stream == NULL) {
+			if (headsets[i].headset_conn == conn) {
+				*index = i;
+				return 0;
+			}
+
+		} else if (conn == NULL) {
+			if (headsets[i].sink_stream == stream) {
+				*index = i;
+				return 0;
+			}
+
+			if (headsets[i].source_stream == stream) {
```

Comment:
Can do "else if" here

Reply:
Hmm, we still have to return if there is a match and I think it doesn't make sense to do:
```C
if (headsets[i].sink_stream == stream) {
        *index = i;
	return 0;
} else if (headsets[i].source_stream == stream) {
	*index = i;
	return 0;
}
```

Reply:
I guess I could do:
```C
if (headsets[i].sink_stream == stream || headsets[i].source_stream == stream) {
	*index = i;
	return 0;
}
```

Reply:
FYI, I did the latter